### PR TITLE
Queues - Hyunji Kim - ride-share-two

### PR DESCRIPTION
**Pull request by ricecakemonster**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? |  The design decision I had to make : Which file should be created first among three class files?  Considering which file would be in the middle of the process in terms of sending and receiving info helped me to decide.      |
| Describe a concept that you gained more clarity on as you worked on this assignment. |  How to make files(classes) work together without colliding.  |
| Describe a _nominal test_ that you wrote for this assignment. |  Making sure a class creates a class instance.  |
| Describe an _edge case test_ that you wrote for this assignment. | In case there're no trips taken by a rider, and this rider's trip info is requested.     |
| How do you feel you did in writing pseudocode first, then writing the tests and then the code? | Helped me to organize the thought process. It was very easy to know what to do next.   It's a very neat way to work!  |

### CODE REVIEW

File: lib/ride_share_driver.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        csv.push({id: driver[0].to_i, name: driver[1], vin: driver[2]})
+      end
+      return csv
+    end
+
+#self.method2 : find a specific driver using their numeric ID
+    def self.find(id_num)
+      raise ArgumentError.new("Not a valid ID number") if id_num.class != Integer
+      result  = all.find {|driver| driver[:id] == id_num}
+      result ||= "No match"
```
Comment (droberts-sea) [inline lib/ride_share_driver.rb:33]: I like the idea of returning a sentinel value here if the requested driver doesn't exist. However, `nil` is usually a better choice than a string - that's basically why it's part of the language!

File: specs/ride_share_dirver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+end
+
+describe "Driver#average_rating" do
+  it "Retrieve an average rating for that driver based on all trips taken" do
+    args = {id: 1, name: "Bernardo Prosacco", vin: "WBWSS52P9NEYLVDE9"}
+    driver = Rideshare::Driver.new(args)
+    driver.average_rating.must_equal 2.33
+  end
+
+  it "In case there's no average rating for driver, it tells so." do
```
Comment (droberts-sea) [inline specs/ride_share_dirver_spec.rb:75]: Good test case! Edge cases like this, asking what happens when there *isn't* any data, can be easy to miss.

**Comment by droberts-sea** [comment]
# Ride Share
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
|  **Baseline** |   |
|   Used Git Regularly	|   Good progress here. I would still like to see more descriptive commit messages - instead of "adding more edge cases", provide a short description of what you've tested. |
|   Answer comprehension questions	|   yes	|
|  **Driver** |   |
|   Uses the `all` method in the `find` method | yes |
|   Has appropriate edge-case tests for each method in the class |   yes |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips | yes |
|   Created a method that uses the internal trips list to calculate the average rating | yes |
|  **Rider** |   |
|   Uses the `all` method in the `find` method | yes |
|   Has appropriate edge-case tests for each method in the class |   yes |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips | yes |
|   Created a method that uses the internal trips method to retrieve the associated drivers | yes |
|  **Trip** |   |
|   Reads the CSV file in the `all` method | yes |
|   Has appropriate edge-case tests for each method in the class |   yes |
|   Created a method that uses a method from the `Driver` to retrieve the associated driver instance  |   yes |
|   Created a method that uses a method from the `Rider` to retrieve the associated rider instance |   yes |
|   Created a method to retrieve all trips by driver id |    yes |
|   Created a method to retrieve all trips by rider id |     yes |
|   **Overall**	|   	|

Small nitpick: Since we're already in the `ride_share` project, you don't need to prefix your file names with `ride_share_`.

Great job! Code is clear and readable, tests cover everything I was looking for, and git habits appear to be headed in the right direction. Keep up the good work!
